plugins/ExtendedTools/objprp.c

```diff
@@ -887,6 +966,54 @@ PPH_STRING EtGetWindowStationType(
     return format[0].u.Char != UNICODE_NULL ? PhFormat(format, RTL_NUMBER_OF(format), 0) : NULL;
 }
 
+ULONG EtSessionIdFromObjectName(
+    _In_ PPH_STRINGREF Name
+    )
+{
+    ULONG sessionId = ULONG_MAX;
+
+    if (PhCountStringZ(Name->Buffer) > RTL_NUMBER_OF(L"Session") - 1)
+    {
+        PH_STRINGREF idString;
+        LONG64 id;
+        PhInitializeStringRef(&idString, &Name->Buffer[RTL_NUMBER_OF(L"Session") - 1]);
+        if (PhStringToInteger64(&idString, 0, &id))
+            sessionId = (ULONG)id;
+    }
```

Comment:
This seems fine, but maybe `PhSplitStringRefAtString` would be better?

Reply:
Oops, this was that easy :relaxed: